File: pbwebrce/pbwebrce-jar/src/main/java/it/csi/pbandi/pbwebrce/integration/request/SalvaRimodulazioneRequest.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbwebrce.integration.request;

import java.util.ArrayList;

import it.csi.pbandi.pbwebrce.dto.ContoEconomicoItem;

public class SalvaRimodulazioneRequest {
	
	private Long idProgetto;
	
	private ArrayList<ContoEconomicoItem> contoEconomicoItemList;
	
	public Long getIdProgetto() {
		return idProgetto;
	}
	public void setIdProgetto(Long idProgetto) {
		this.idProgetto = idProgetto;
	}
	public ArrayList<ContoEconomicoItem> getContoEconomicoItemList() {
		return contoEconomicoItemList;
	}
	public void setContoEconomicoItemList(ArrayList<ContoEconomicoItem> contoEconomicoItemList) {
		this.contoEconomicoItemList = contoEconomicoItemList;
	}
	
}

File: pbwebrcewcl/src/app/rimodulazione-conto-economico/commons/request/salva-rimodulazione-request.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { VociDiEntrata } from "src/app/rimodulazione-conto-economico-a20/commons/conto-economico-vo";
import { AltriCostiDTO } from "../dto/altri-costi-dto";
import { ContoEconomicoItem } from "../dto/conto-economico-item";

export class SalvaPropostaRimodulazioneRequest {
    constructor(
        public idProgetto: number,
        public contoEconomicoItemList: Array<ContoEconomicoItem>,
        public altriCosti?: Array<AltriCostiDTO>,
    ) { }
}

export class SalvaRimodulazioneRequest extends SalvaPropostaRimodulazioneRequest { }